### ISSUE TITLE
Output truncated in the extension

### ISSUE THREAD

**Issue by pums974** [continuedev/continue#793]
### Before submitting your bug report

- [X] I believe this is a bug. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [X] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue) that reports the same bug
- [X] I've seen the [troubleshooting guide](https://continue.dev/docs/troubleshooting) on the Continue Docs

### Relevant environment info

```Markdown
- OS: Arch Linux
- Continue: 0.8.5
- IDE: VSCode
```

### Description

While the model (codellama:7b) answered (badly) to my prompt, and the transcript shows the entirety of it (see bellow) the interface show almost no output. 

This might be caused by a block of code of a language not supported by the markdown interpreter ?

![image](https://github.com/ollama/ollama/assets/1005109/8903d086-a2cd-4875-bc96-8c9505bda1d7)

# Transcript

This is a session transcript from [Continue](https://continue.dev) on 1/29/2024, 3:06:56 PM.

## User

```truc.py (28-35)
def mysterious_function(x):
    for i in range(len(x)):
        for j in range(len(x) - 1):
            if x[j] > x[j + 1]:
                x[j], x[j + 1] = x[j + 1], x[j]

    return x

```
rewrite this python function in fortran

## Continue

Here is one way to rewrite the Python function `mysterious_function` in Fortran:
```fortran
subroutine mysterious_function(x)
  integer :: i, j
  real :: x(n)
  
  do i = 1, n - 1
    do j = i + 1, n
      if (x(j) > x(i)) then
        call swap(x(i), x(j))
      end if
    end do
  end do
  
  return
end subroutine
```
Here's a breakdown of the Fortran code:

* The `subroutine` keyword is used to define a new subroutine.
* The `mysterious_function` subroutine takes a single argument `x`, which is an array of real numbers with length `n`.
* The `do` loop iterates over the elements of the `x` array, starting from index 1 and ending at index `n - 1`.
* The inner `do` loop iterates over the elements of the `x` array starting from index `i + 1` and ending at index `n`.
* The `if` statement checks if the element at index `j` is greater than the element at index `i`. If it is, then we swap the two elements using the `swap` function.
* The `return` keyword is used to exit the subroutine and return control to the calling program.

Note that in Fortran, arrays are 1-based, so the first element of an array has index 1, not 0. Also, Fortran uses explicit typing for variables, so we need to specify the type of `x` as a real number array with length `n`.

## User

/share

**Comment by sestinj** [comment]
@pums974 thanks for all the detail, this cleanly reproduces and seems to be caused by Shiki for only certain languages, or potentially unlabeled code blocks. will update as soon as I have a solution

**Comment by sestinj** [comment]
@pums974 Your guess was exactly right, and just removing the language from the code block if unsupported (or replacing with "javascript" to at least get partial highlighting) seemed to solve it. Changes here; https://github.com/continuedev/continue/commit/ca30daf5a93a76f490b4764d209bcf077538417a

This is now available in pre-release version 0.9.38. If you can confirm that it looks good then I think this issue can be closed. Regarding full support for fortran or other languages not already included, this might be best in a separate issue

**Comment by pums974** [comment]
Hi, I just tried the pre-release version v0.9.42 and it works perfectly.
Thank you four your reactivity
